modules/local/locidex_select.nf

```diff
@@ -138,57 +140,86 @@ def select_locidex_db_path(db_values, db_name){
-            exit 1, "Missing config data for locidex database entry: ${db_name}"
+        if(!db_entry.containsKey(params.locidex.manifest_config_key)){
+            error("Missing config data for locidex database entry: ${db_name}")
         }
-        if(!value[params.locidex.manifest_config_key].containsKey(params.locidex.database_config_value_date)){
-            exit 1, "Missing date created value for locidex database entry: ${db_name}"
+        if(!db_entry[params.locidex.manifest_config_key].containsKey(params.locidex.manifest_config_name)){
+            error ("Missing name value in locidex config for: ${db_name}")
         }
-        def date_value = value[params.locidex.manifest_config_key][params.locidex.database_config_value_date]
-        def date_check = new SimpleDateFormat(params.locidex.date_format_string).parse(date_value)
-        dates << date_check
+        if(!db_entry[params.locidex.manifest_config_key].containsKey(params.locidex.database_config_value_date)){
+            error("Missing date created value for locidex database entry: ${db_name}")
+        }
+        def date_value = db_entry[params.locidex.manifest_config_key][params.locidex.database_config_value_date]
+        def date_check = null
+        try{
+            date_check = new SimpleDateFormat(params.locidex.date_format_string).parse(date_value)
+        }catch (java.text.ParseException e){
+            error("Date value ${date_value} does not meet format string required of ${params.locidex.date_format_string}")
+        }
+
+        dates.add(date_check)
         if(date_check > max_date){
             max_date = date_check
-            max_date_idx = idx
+            max_date_entry = db_entry
         }
-        idx += 1
     }
 
     def max_date_count = dates.count(max_date)
     if(max_date_count > 1){
-        exit 1, "There are multiple versions of the most recent database for ${db_name}. Mikrokondo could not determine the best database to pick."
+        error("There are multiple versions of the most recent database for ${db_name}. Mikrokondo could not determine the best database to pick.")
+    }else if (max_date_count == 0){
+        error("There are not databases created after the year ${default_date}. Please set the allele database parameter, or adjust the date your database was created in the 'config.json'")
     }
-    return db_values[max_date_idx]
+
+    return max_date_entry
 }
 
 
-def tokenize_values(species, match_size){
-    // Tokenize the species values to find the right match
-    def tokens = []
+def window_string(species, match_size){
+    /*
+        Create an array of strings of a various match "match size" for comparison to a given value later one.
+
+        e.g. spieces is an array of: ["1", "2", "3", "4"] and match_size is 2 the output will be.
+            [
+                ["1", "2"],
+                ["2", "3"],
+                ["3", "4"]
+            ]
+    */
+    def tiles = []
```

Comment:
This seems similar in concept to a sliding window when working with k-mers. k-merization, I think, but the word won't make sense here.

Maybe something like: "Generates all array slices of length `match_size` from the `species` array."? I think keeping the example you have here is useful for illustrating this.

Reply:
Indeed it is quite similar! 

Originally I was thinking of using the `Damerau-Levenshtein` distance for indentifying the best match. But I figured typos should be punished in this instance.

Also changed the function comment to what you have recommended.

Reply:
fixed in: [4883b39](https://github.com/phac-nml/mikrokondo/pull/96/commits/4883b39c98601dc071a971c765b978f622a19a1c)